File: common/src/test/java/guepardoapps/lucahome/common/services/UserService/IsValidUserNameUnitTest.java
package guepardoapps.lucahome.common.services.UserService;

import guepardoapps.lucahome.common.services.user.UserService;

import org.junit.experimental.theories.DataPoints;
import org.junit.experimental.theories.Theories;
import org.junit.experimental.theories.Theory;
import org.junit.runner.RunWith;

import static org.junit.Assert.assertEquals;

@RunWith(Theories.class)
public class IsValidUserNameUnitTest {

    public static class UserNameValidation {
        String userName;
        boolean isValid;

        UserNameValidation(String userName, boolean isValid) {
            this.userName = userName;
            this.isValid = isValid;
        }
    }

    @DataPoints
    public static UserNameValidation[] userNames = {
            new UserNameValidation("Administrator", true),
            new UserNameValidation("Admin1", false),
            new UserNameValidation("Admin.", false),
            new UserNameValidation("Adm", false),
            new UserNameValidation("12345", false),
            new UserNameValidation("..--..", false),
            new UserNameValidation("Test", true),
            new UserNameValidation("Tester", true),
            new UserNameValidation("", false),
            new UserNameValidation("TesterHelloWorld", true),
            new UserNameValidation("TesterHelloWorldI", false)
    };

    @Theory
    public void isValidUserName_isCorrect(UserNameValidation userNameValidation) throws Exception {
        boolean actualIsValid = UserService.Companion.getInstance().isValidUserName(userNameValidation.userName);
        assertEquals(userNameValidation.isValid, actualIsValid);
    }
}

File: common/src/test/java/guepardoapps/lucahome/common/services/user/IsValidUserNameUnitTest.kt
package guepardoapps.lucahome.common.services.user

import org.junit.experimental.theories.DataPoints
import org.junit.experimental.theories.Theories
import org.junit.experimental.theories.Theory
import org.junit.runner.RunWith

import org.junit.Assert.assertEquals

@RunWith(Theories::class)
class IsValidUserNameUnitTest {

    class UserNameValidation internal constructor(internal var userName: String, internal var isValid: Boolean)

    companion object {

        @DataPoints
        var userNames = arrayOf(
                UserNameValidation("Administrator", true),
                UserNameValidation("Admin1", false),
                UserNameValidation("Admin.", false),
                UserNameValidation("Adm", false),
                UserNameValidation("12345", false),
                UserNameValidation("..--..", false),
                UserNameValidation("Test", true),
                UserNameValidation("Tester", true),
                UserNameValidation("", false),
                UserNameValidation("TesterHelloWorld", true),
                UserNameValidation("TesterHelloWorldI", false)
        )
    }

    @Theory
    @Throws(Exception::class)
    fun isValidUserName_isCorrect(userNameValidation: UserNameValidation) {
        val actualIsValid = UserService.instance.isValidUserName(userNameValidation.userName)
        assertEquals(userNameValidation.isValid, actualIsValid)
    }
}